hoodie-client/src/main/java/com/uber/hoodie/io/HoodieMergeHandle.java

```diff
@@ -71,30 +72,33 @@ public HoodieMergeHandle(HoodieWriteConfig config, String commitTime, HoodieTabl
       Iterator<HoodieRecord<T>> recordItr, String fileId) {
     super(config, commitTime, hoodieTable);
     this.fileSystemView = hoodieTable.getROFileSystemView();
-    init(fileId, init(fileId, recordItr));
+    String partitionPath = init(fileId, recordItr);
+    init(fileId, partitionPath,
+        fileSystemView.getLatestDataFiles(partitionPath)
+            .filter(dataFile -> dataFile.getFileId().equals(fileId)).findFirst());
   }
 
   public HoodieMergeHandle(HoodieWriteConfig config, String commitTime, HoodieTable<T> hoodieTable,
-      Map<String, HoodieRecord<T>> keyToNewRecords, String fileId) {
+      Map<String, HoodieRecord<T>> keyToNewRecords, String fileId, Optional<HoodieDataFile> oldDataFileToBeMerged) {
```

Comment:
We should avoid passing an extra HoodieDataFile here.

Reply:
this is in the same spirit as Vinoth explained